client/data/settings/actions.js

```diff
@@ -13,6 +13,28 @@ import { __ } from '@wordpress/i18n';
 import ACTION_TYPES from './action-types';
 import { NAMESPACE, STORE_NAME } from '../constants';
 
+function setSettingsValue( payload ) {
+	return {
+		type: ACTION_TYPES.SET_SETTINGS_VALUE,
+		payload,
+	};
+}
```

Comment:
Good point on the `update*` name - I was using the action type name, which is `SET_SETTINGS_VALUE`/`SET_SETTINGS`. Maybe even the action (or action type) in `updateSettings` (and all the other actions) should also be renamed? 👀 

In the meantime, I renamed this to `updateSettingsValues`. A more complete renaming can definitely be done later 👍 

I'd like to keep it more "open", with the `payload` object, so that multiple values can be passed 👍

Reply:
> Maybe even the action (or action type) in updateSettings (and all the other actions) should also be renamed? 👀

I'm not sure about this. I noticed all of the other data modules in WCPay use this convention so that'd likely be a huge refactor, unless everyone is on board, of course!

> In the meantime, I renamed this to `updateSettingsValues`. A more complete renaming can definitely be done later 👍
> 
> I'd like to keep it more "open", with the `payload` object, so that multiple values can be passed 👍
> 
> One cool thing about the object approach as argument, is that you can specifically type the object in typescript as a `Partial<State>` and you would have TS validation on it. Whereas if the arguments were the property's name and the property's value, you wouldn't be able to infer the type of the property value by its name - I'm not sure how we could do that in TS 🤔

Wow, awesome! And TIL!